tools/lint_diff.py

```diff
@@ -0,0 +1,84 @@
+import os
+import sys
+import subprocess
+
+CONFIG = os.path.join(
+    os.path.abspath(os.path.dirname(__file__)),
+    'lint_diff.ini',
+)
+
+
+def rev_list(branch, num_commits):
+    """List commits in reverse chronological order.
+
+    Only the first `num_commits` are shown.
+
+    """
+    res = subprocess.run(
+        [
+            'git',
+            'rev-list',
+            '--max-count',
+            f'{num_commits}',
+            '--first-parent',
+            branch
+        ],
+        capture_output=True,
+        text=True,
+    )
+    res.check_returncode()
+    return res.stdout.rstrip('\n').split('\n')
+
+
+def find_branch_point():
+    """Find when the current branch split off master.
+
+    It is based off of this Stackoverflow post:
+
+    https://stackoverflow.com/questions/1527234/finding-a-branch-point-with-git#4991675
+
+    """
+    branch_commits = rev_list('HEAD', 100)
+    master_commits = set(rev_list('master', 100))
```

Comment:
IIUC this is looking back 100 commits on the branch.  Do we have stats on the number of commits per PR? My expectation is that 100 is reasonable, but maybe there have been some PRs with a lot of commits??
And also looking back on 100 commits on master.  I wonder how many PRs branched more than 100 commits ago, and yet don't need a rebase to avoid conflicts? This actually does seem possible given the elapsed time to review and merge PRs.

Reply:
> It seems like a reasonable number but it might be that we want to make that adjustable

I don't think we want it to be adjustable-this should be a simple tool that "just works". I bumped it up to 1000, if that's not enough we can increase it further. The only thing it's really trying to avoid is making a set of all 22,000 commits.

Reply:
> wonder how many PRs branched more than 100 commits ago, and yet don't need a rebase to avoid conflicts?

How would the conflicts impact the linter? Not clear to me. 

>I don't think we want it to be adjustable-this should be a simple tool that "just works".

I agree I had misunderstood what was happening in the code here, looks like the 100 commits should be reasonable on the 'HEAD' branch. If the 100 is too much on the master then the person should rebase-is this the idea/intent?
New committers often have difficulty we using the rebase command. 

>The only thing it's really trying to avoid is making a set of all 22,000 commits.

I agree that 22K is not needed, it does seem like more would be needed on master than on the branch itself to execute without error though.